client/blocks/upload-image/index.jsx

```diff
@@ -0,0 +1,199 @@
+/**
+ * External dependencies
+ */
+import React, { Component, PropTypes } from 'react';
+import classnames from 'classnames';
+import { connect } from 'react-redux';
+import { localize } from 'i18n-calypso';
+import path from 'path';
+import Gridicon from 'gridicons';
+import { noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { ALLOWED_FILE_EXTENSIONS } from './constants';
+import { AspectRatios } from 'state/ui/editor/image-editor/constants';
+import Dialog from 'components/dialog';
+import FilePicker from 'components/file-picker';
+import {
+	resetAllImageEditorState
+} from 'state/ui/editor/image-editor/actions';
+import Spinner from 'components/spinner';
+import ImageEditor from 'blocks/image-editor';
+import DropZone from 'components/drop-zone';
+
+class UploadImage extends Component {
+	state = {
+		isEditingImage: false,
+		uploadedImage: null,
+		editedImage: null,
+	};
+
+	static propTypes = {
+		isUploading: PropTypes.bool,
+		// Additional props passed to ImageEditor component. See blocks/image-editor.
+		imageEditorProps: PropTypes.object,
+		texts: PropTypes.object,
+		onImageEditorDone: PropTypes.func,
+		additionalImageEditorClasses: PropTypes.string,
+		additionalClasses: PropTypes.string,
+	};
+
+	static defaultProps = {
+		imageEditorProps: {
+			allowedAspectRatios: [ AspectRatios.ASPECT_1X1 ],
+		},
+		texts: {},
+		backgroundContent: null,
+		onImageEditorDone: noop,
+		isUploading: false,
+	};
+
+	onReceiveFile = ( files ) => {
```

Comment:
the name of the method shouldn't be as an event. For instance, a valid name IMO could be `receiveFiles` (plural), because that's what the method will do when it's invoked. 

```jsx
<FilePicker accept="image/*" onPick={ this.receiveFiles }>
```

```es6
receiveFiles = ( files ) => {
  // ...
}
```